### PR TITLE
🚀  Add Live Help Feature & Loading States

### PR DESCRIPTION
**Pull request by chase-manning**
# Overview
This PR makes a few changes and additions:
* Introduces a new Live Help feature to give suggestions to the user
* Moves more business logic into selectors and out of components
* Changes unloaded data to return as null instead of `0`
* Introduces a loading state for most components
* Added Pool TVL to pool statistics

## Live Help
Suggestions have been added throughout the user flow when certain criteria are met.  
Details of these suggestions and what triggers them can be [found here](https://docs.google.com/document/d/10bnRnIUvvSDDlDdfPXBMyefBVxu5Wz10KOEtK1MaJgk/edit#).
image

## Loading State
Previously most of the state that required loading would return a default value before being loaded.  
For example, the users balance of a given token would start at `0` and then be set to the actual value once loaded.  
This meant that the users could not tell if a user actually had no balance for a token, or if it was still loading.  
It also resulted in flickering on the screen as various values flicker between $0, then to maybe $10 when part of a value is loaded, then maybe finally to $700.  
To resolve this, all loadable states have been changed to return null by default.  
All components have also been changed to show a new Loader when the value they render is null.

Another benefit of this approach is that by supporting an empty state, we can now directly route to pool pages.  
Previously if you navigated to `https://backd.fund/pool/bDAI` it would redirect you to the landing page, because that pool didn't exist as it isn't loaded yet.  
Now it will instead load the page in the empty loaded state first, then wait until the pools have been fetched, then only redirect if the pools have been fetched and it also cannot find that pool, so it only only redirects to landing page for non-existent pools.
image
image
image

## Pool TVL
Pool TVL was added to the pool statistics:
image

### CODE REVIEW

**Comment by danhper** [comment]
Task linked: [CU-7h2h63 Show Loading Indicator for while data loads from blockchian](https://app.clickup.com/t/7h2h63)

**Comment by danhper** [comment]
Task linked: [CU-c00cpe Move tvl logic into selector](https://app.clickup.com/t/c00cpe)

**Comment by danhper** [comment]
Task linked: [CU-c5pmfq Add Pool TVL to Statistics](https://app.clickup.com/t/c5pmfq)

**Comment by danhper** [comment]
Task linked: [CU-1q33za6 Add Guidance to Topup Field Entries](https://app.clickup.com/t/1q33za6)

**Comment by danhper** [comment]
Task linked: [CU-axt4mb Add Support for direct navigation to pool](https://app.clickup.com/t/axt4mb)

File: src/components/LiveHelp.tsx [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+  useEffect(() => {
+    setOpen(hasSuggestions);
+  }, [hasSuggestions]);
+
+  useEffect(() => {
+    if (hasLowPositions) {
+      lowPositions.forEach((position: Position) =>
+        dispatch(
+          addSuggestion({
```
[the remark this answers is not in the document]
Comment (chase-manning) [reply src/components/LiveHelp.tsx:218]: Good idea!

File: src/components/LiveHelp.tsx [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+            data: position.protocol.toLowerCase(),
+            text: t("liveHelp.suggestions.topupPositionLow.text", { protocol: position.protocol }),
+            button: t("liveHelp.suggestions.topupPositionLow.button"),
+            link: "https://docs.backd.fund/protocol-architecture/actions/top-ups",
+          })
+        )
+      );
+      return;
+    }
+    dispatch(removeSuggestion(Suggestion.POSITION_LOW));
```
[the remark this answers is not in the document]
Comment (chase-manning) [reply src/components/LiveHelp.tsx:229]: True that is nice! ty ty

File: src/components/LiveHelp.tsx [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+    if (
+      implement &&
+      implement.type === Suggestion.POSITION_LOW &&
+      location.pathname !== "/actions"
+    ) {
+      navigate("/actions");
+    }
+  }, [implement, location]);
+
+  if (!hasSuggestions) return <div />;
```
[the remark this answers is not in the document]
Comment (chase-manning) [reply src/components/LiveHelp.tsx:242]: I prefer the `null` also! Have updated this, and also all other components where we were returning `<div />`.

File: src/pages/actions/YourDeposits.tsx [hunk 4]
```diff
@@ -128,37 +121,42 @@ const YourDeposits = (): JSX.Element => {
...
-                  </Balances>
-                </Row>
-              ))}
-              <Total id="your-deposits-total">{`= ${formatCurrency(
-                Number(totalUsd.toString())
-              )}`}</Total>
+              {depositedPools.map((pool: Pool) => {
+                const price = prices[pool.underlying.symbol];
+                const balance = balances[pool.lpToken.address];
+                return (
```
[the remark this answers is not in the document]
Comment (chase-manning) [reply src/pages/actions/YourDeposits.tsx:132]: Ohh yeah that's much nicer! Thanks! 🙏🏻

**Pull request merged**